### PR TITLE
✨ Ask user to upgrade patch versions for services

### PR DESCRIPTION
**Pull request by jsaq007**
## What do these changes do?

Auto Upgrade of patch versions #1229

Layout issue #5240

## Related issue/s

- resolves Auto Upgrade of patch versions #1229
- fixes Layout issue #5240
- Resolves - Dashboard list views have low contrast #[1234](https://github.com/ITISFoundation/osparc-issues/issues/1234)
- Resolves - New feedback: Add Editors... [190346](https://z43.fogbugz.com/f/cases/190346)

## Dev Checklist

- [x] No ENV changes or I properly updated ENV ([read the instruction](https://git.speag.com/oSparc/osparc-ops-deployment-configuration/-/blob/configs/README.md?ref_type=heads#how-to-update-env-variables))

### CODE REVIEW

**Review by matusdrobuliak66 — APPROVED** [review]

**Review by mguidon — COMMENTED** [review]
Thank you for this! So I understand correctly that it is now no longer possible to open without upgrading? I am fine with that, but lets see how users react if there is a bug in the newer version.

**Review by mguidon — APPROVED** [review]

**Comment by jsaq007** [comment]
> Thank you for this! So I understand correctly that it is now no longer possible to open without upgrading? I am fine with that, but lets see how users react if there is a bug in the newer version.

if the user cancels they go into the study. I Plan to further this and make the two options more clear

File: services/static-webserver/client/source/class/osparc/dashboard/ResourceMoreOptions.js [hunk 1]
```diff
@@ -160,10 +190,39 @@ qx.Class.define("osparc.dashboard.ResourceMoreOptions", {
...
-      });
-      toolbar.add(openButton);
+      }, true);
+    },
 
-      this._add(toolbar);
+    __confirmUpdate: function() {
+      this.__openButton.setLabel("Update");
+      this.__openButton.addListenerOnce("tap", () => {
+        const msg = this.tr("Some of your services are outdated. For imploved performance please update to the latest version.");
```
Comment (elisabettai) [inline services/static-webserver/client/source/class/osparc/dashboard/ResourceMoreOptions.js:199]: There's a typo here "imploved"

**Review by elisabettai — APPROVED** [review]
Thanks a lot!

It is nice that they can still choose do don't upgrade (while for sim4life.io it's preferable that they update, for osparc.io I would not enforce it). 

It is not clear whether they can choose not to update. I would:
1. Rename the "Cancel" button to "Stay outdated" (or something more professional) or
2. Clarify in the message that clicking on "Cancel" will result in keeping the current version

File: services/static-webserver/client/source/class/osparc/dashboard/ResourceMoreOptions.js (hunk 1 above)
Comment (jsaq007) [reply services/static-webserver/client/source/class/osparc/dashboard/ResourceMoreOptions.js]: @elisabettai I improved the text with.
Some of your services are outdated. For imploved performance please update to the latest version.
What do you think?

**Review by pcrespov — COMMENTED** [review]
:+1:

**Review by sanderegg — APPROVED** [review]
very nice! thanks

**Review by mguidon — APPROVED** [review]

**Review by elisabettai — APPROVED** [review]

**Review by ignapas — COMMENTED** [review]
How can I start the study without updating?

**Comment by ignapas** [comment]
> How can I start the study without updating?

Ok. I see that after clicking on "Update" the user can still choose not to update, but that I find confusing. I would keep the text on the button to "Open" and then ask the user whether he wants to update or not

**Comment by jsaq007** [comment]
> > How can I start the study without updating?
> 
> Ok. I see that after clicking on "Update" the user can still choose not to update, but that I find confusing. I would keep the text on the button to "Open" and then ask the user whether he wants to update or not

I like that... do you think it's clear enough there are updates or is it necessary for the user to know at that point? But I like your thinking I will do it.

**Review by GitHK — APPROVED** [review]

**Comment by ignapas** [comment]
> > > How can I start the study without updating?
> > 
> > 
> > Ok. I see that after clicking on "Update" the user can still choose not to update, but that I find confusing. I would keep the text on the button to "Open" and then ask the user whether he wants to update or not
> 
> @ignapas I like that... do you think it's clear enough there are updates or is it necessary for the user to know at that point? But I like your thinking I will do it.

Well my reasoning was that if there are updates, there is no Open button anymore! It should always be there imo. But I agree: it would probably be nice to have a small text close to the button or a tooltip saying that there are updates available (see what looks/feels best), and then the user could get asked after he clicks Open, like it happens now.
For the release/master branches mess, maybe @mguidon could also take a look? :D

**Review by ignapas — APPROVED** [review]
After the Open button is back and master is taken out 👍🏼

**Comment by mguidon** [comment]
> > > > How can I start the study without updating?
> > > 
> > > 
> > > Ok. I see that after clicking on "Update" the user can still choose not to update, but that I find confusing. I would keep the text on the button to "Open" and then ask the user whether he wants to update or not
> > 
> > 
> > @ignapas I like that... do you think it's clear enough there are updates or is it necessary for the user to know at that point? But I like your thinking I will do it.
> 
> Well my reasoning was that if there are updates, there is no Open button anymore! It should always be there imo. But I agree: it would probably be nice to have a small text close to the button or a tooltip saying that there are updates available (see what looks/feels best), and then the user could get asked after he clicks Open, like it happens now. For the release/master branches mess, maybe @mguidon could also take a look? :D

Master/release mess?

**Comment by ignapas** [comment]
> > > > > > How can I start the study without updating?
> > > > > 
> > > > > 
> > > > > Ok. I see that after clicking on "Update" the user can still choose not to update, but that I find confusing. I would keep the text on the button to "Open" and then ask the user whether he wants to update or not
> > > > 
> > > > 
> > > > @ignapas I like that... do you think it's clear enough there are updates or is it necessary for the user to know at that point? But I like your thinking I will do it.
> > > 
> > > 
> > > Well my reasoning was that if there are updates, there is no Open button anymore! It should always be there imo. But I agree: it would probably be nice to have a small text close to the button or a tooltip saying that there are updates available (see what looks/feels best), and then the user could get asked after he clicks Open, like it happens now. For the release/master branches mess, maybe @mguidon could also take a look? :D
> > 
> > 
> > Master/release mess?
> 
> yes, master was accidentally merged here, but I think it is solved now

but we are in simcore! it is master that you have to merge here so that is ok... oder? :S

**Pull request merged**